src/main/java/org/opensearch/neuralsearch/ml/MLCommonsClientAccessor.java

```diff
@@ -144,4 +175,20 @@ private List<List<Float>> buildVectorFromResponse(MLOutput mlOutput) {
         return vector;
     }
 
+    private List<Map<String, ?>> buildMapResultFromResponse(MLOutput mlOutput) {
+        final ModelTensorOutput modelTensorOutput = (ModelTensorOutput) mlOutput;
```

Comment:
Same as the above one. Maybe we can use consistent code between the 2 use case. And If they need to be fixed, we can create another PR to fix them. https://github.com/opensearch-project/neural-search/blame/8484be9b319a09dbdd9a1c952ca1b8515dedf571/src/main/java/org/opensearch/neuralsearch/ml/MLCommonsClientAccessor.java#L136

Reply:
We can fix the other one in the future, but I want to understand how we know for certain this wont cause class cast on invalid input.

Reply:
@zane-neo could you please help answer this?

Reply:
The two inference method we created in neural-search(inferenceWithVectorResult & inferenceWithMapResult) are for text-embedding and sparse-embedding only, and they both return ModelTensorOutput. For later one in the future, it could be expanded to other use cases, e.g. remote inference, but in this case, it's still returning ModelTensorOutput. So it's safe to cast this output to ModelTensorOutput.